### PR TITLE
Fix EBSVolumeHasSSERule for bool type

### PR DESCRIPTION
**Pull request by w0rmr1d3r**
### Fixes
- `EBSVolumeHasSSERule` can now understand `encrypted_status` if modelled as a `bool`.
### Updates
- Updated `EBSVolumeHasSSERule` to iterate only over `AWS::EC2::Volume` resources.

### CODE REVIEW

File: tests/rules/test_EBSVolumeHasSSERule.py [hunk 1]
```diff
@@ -25,9 +25,12 @@ def test_no_failures_are_raised(good_template):
     assert compare_lists_of_failures(result.failures, [])
 
 
-def test_failures_are_raised(bad_template):
+@pytest.mark.parametrize(
```
Comment (jsoucheiron) [inline tests/rules/test_EBSVolumeHasSSERule.py:28]: Why are we using parametrize here?

File: cfripper/rules/ebs_volume_has_sse.py [hunk 2]
```diff
@@ -47,21 +47,15 @@ class EBSVolumeHasSSERule(Rule):
...
-                            "config": self._config,
-                            "extras": extras,
-                            "logical_id": logical_id,
-                            "resource": resource,
-                        },
-                    )
+        for logical_id, resource in cfmodel.resources_filtered_by_type("AWS::EC2::Volume").items():
+            encrypted_status = getattr(resource.Properties, "Encrypted", None)
+
+            if encrypted_status is None or encrypted_status is False or encrypted_status.lower() != "true":
```
Comment (jsoucheiron) [inline cfripper/rules/ebs_volume_has_sse.py:53]: This is dangerous:
image

```suggestion
            if str(encrypted_status).lower() != "true":
```

File: tests/rules/test_EBSVolumeHasSSERule.py (hunk 1 above)
Comment (jsoucheiron) [reply tests/rules/test_EBSVolumeHasSSERule.py:28]: I completely missed that the list has 2 entries, I only saw one :( #FAIL

**Review by jsoucheiron — APPROVED** [review]

**Review by oscarbc96 — CHANGES_REQUESTED** [review]
let's wait for https://github.com/Skyscanner/pycfmodel/pull/95 to be merged

**Review by ocrawford555 — APPROVED** [review]

**Pull request merged**